fda-model/structures/Inventory.cs

```diff
@@ -0,0 +1,203 @@
+            where T : struct
+        {
+            if (value == null)
+                return defaultValue;
+            else if (value == DBNull.Value)
+                return defaultValue;
+            else
+            {
+                var retn = value as T?;
+                if (retn.HasValue)
+                    return retn.Value;
+                else
+                    return defaultValue;
+            }
+        }
+        public static T TryGetObj<T>(object value, T defaultValue = default)
+            where T : class
+        {
+            if (value == null)
+                return defaultValue;
+            else if (value == DBNull.Value)
+                return defaultValue;
+            else
+            {
+                var retn = value as T;
+                if (retn != null)
+                    return retn;
+                else
+                    return defaultValue;
+            }
+        }
+
+
+        /// <summary>
+        /// Constructor to create a SI from a shapefile. Gonna need to do this from database potentially as well
+        /// </summary>
+        /// <param name="pointShapefilePath"></param>
+        public Inventory(string pointShapefilePath, string impactAreaShapefilePath)
+        {
+            PointFeatureLayer structureInventory = new PointFeatureLayer("Structure_Inventory", pointShapefilePath);
+            PointMs pointMs = new PointMs(structureInventory.Points().Select(p => p.PointM()));
+            Structures = new List<Structure>();
+            try
+            {
+                for (int i = 0; i < structureInventory.FeatureCount(); i++)
+                {
+                    PointM point = pointMs[i];
+                    var row = structureInventory.FeatureRow(i);
+                    int fid = TryGet<int>(row["fd_id"]);
+                    double found_ht = TryGet<double>(row["found_ht"]);
+                    double ground_elv = TryGet<double>(row["ground_elv"]);
+                    double val_struct = TryGet<double>(row["val_struct"]);
+                    double val_cont = TryGet<double>(row["val_cont"]);
+                    double val_vehic = TryGet<double>(row["val_vehic"]);
+                    double val_other = TryGet<double>(row["val_vehic"]);
+                    string st_damcat = TryGetObj<string>(row["st_damcat"]);
+                    string occtype = TryGetObj<string>(row["occtype"]);
+                    string cbfips = TryGetObj<string>(row["cbfips"]);
+                    double ff_elev = TryGet<double>(row["ff_elev"]);
+                    if (row["ff_elev"] == System.DBNull.Value)
+                    {
+                        ff_elev = ground_elv + found_ht;
+                    }
+                    int impactAreaID = GetImpactAreaID(point, impactAreaShapefilePath);
+                    Structures.Add(new Structure(fid, point, found_ht, val_struct, val_cont, val_vehic, val_other, st_damcat, occtype, impactAreaID, cbfips));
```

Comment:
Note sure if relevant, but the mapping we discussed to avoid the hard-reference to column names is mentioned in this issue: https://github.com/HydrologicEngineeringCenter/HEC-FDA/issues/482

Reply:
fixed. I've created an object to store all the necessary mappings. We'll have the UI create that object for us when we create a structure inventory.

Reply:
awesome!!